handler/src/main/java/io/netty/handler/ssl/SslContext.java

```diff
@@ -673,6 +673,204 @@ static ApplicationProtocolConfig toApplicationProtocolConfig(Iterable<String> ne
+         */
+        public Builder trustManager(File trustCertChainFile) {
+            this.trustCertChainFile = trustCertChainFile;
+            this.trustManagerFactory = null;
+            return this;
+        }
+
+        /**
+         * Trusted manager for verifying the remote endpoint's certificate. Using a {@code
+         * TrustManagerFactory} is only supported for {@link SslProvider#JDK}; for other providers,
+         * you must use {@link #trustManager(File)}. {@code null} uses the system default.
+         */
+        public Builder trustManager(TrustManagerFactory trustManagerFactory) {
+            this.trustCertChainFile = null;
+            this.trustManagerFactory = trustManagerFactory;
+            return this;
+        }
+
+        /**
+         * Identifying certificate for this host. {@code keyCertChainFile} and {@code keyFile} may
+         * be {@code null} for client contexts, which disables mutual authentication.
+         *
+         * @param keyCertChainFile an X.509 certificate chain file in PEM format
+         * @param keyFile a PKCS#8 private key file in PEM format
+         */
+        public Builder keyManager(File keyCertChainFile, File keyFile) {
+            return keyManager(keyCertChainFile, keyFile, null);
+        }
+
+        /**
+         * Identifying certificate for this host. {@code keyCertChainFile} and {@code keyFile} may
+         * be {@code null} for client contexts, which disables mutual authentication.
+         *
+         * @param keyCertChainFile an X.509 certificate chain file in PEM format
+         * @param keyFile a PKCS#8 private key file in PEM format
+         * @param keyPassword the password of the {@code keyFile}, or {@code null} if it's not
+         *     password-protected
+         */
+        public Builder keyManager(File keyCertChainFile, File keyFile, String keyPassword) {
+            if (!forClient) {
+                if (keyCertChainFile == null) {
+                    throw new NullPointerException("keyCertChainFile required for servers");
+                }
+                if (keyFile == null) {
+                    throw new NullPointerException("keyFile required for servers");
+                }
+            }
+            this.keyCertChainFile = keyCertChainFile;
+            this.keyFile = keyFile;
+            this.keyPassword = keyPassword;
+            this.keyManagerFactory = null;
+            return this;
+        }
+
+        /**
+         * Identifying manager for this host. {@code keyManagerFactory} may be {@code null} for
+         * client contexts, which disables mutual authentication. Using a {@code KeyManagerFactory}
+         * is only supported for {@link SslProvider#JDK}; for other providers, you must use {@link
+         * #keyManager(File, File)} or {@link #keyManager(File, File, String)}.
+         */
+        public Builder keyManager(KeyManagerFactory keyManagerFactory) {
+            if (!forClient) {
```

Comment:
We support mutual auth and so I think the client should be able to use this (or some of this) as well.